embeddings/pipeline/preprocessing_pipeline.py

```diff
@@ -52,120 +58,126 @@ def run(self) -> TransformationResult:
+                )
+
+        if self.sample_missing_splits:
             transformation = transformation.then(
-                SampleSplitsFlairCorpusTransformation(*sample_missing_splits, seed=seed)
+                SampleSplitsFlairCorpusTransformation(*self.sample_missing_splits, seed=self.seed)
             )
-        if ignore_test_subset:
-            transformation = transformation.then(DropSubsetFlairCorpusTransformation(subset="test"))
-        transformation = transformation.persisting(FlairPicklePersister(path=persist_path))
-        super().__init__(dataset, data_loader, transformation)
 
+        if self.ignore_test_subset:
+            transformation = transformation.then(DropSubsetFlairCorpusTransformation(subset="test"))
 
-class FlairSequenceLabelingPreprocessingPipeline(
-    PreprocessingPipeline[str, datasets.DatasetDict, Corpus]
-):
-    def __init__(
-        self,
-        dataset_name_or_path: str,
-        persist_path: str,
-        input_column_name: str,
-        target_column_name: str,
-        datasets_path: Path = DATASET_PATH,
-        sample_missing_splits: Optional[Tuple[Optional[float], Optional[float]]] = None,
-        ignore_test_subset: bool = False,
-        seed: int = 441,
-        load_dataset_kwargs: Optional[Dict[str, Any]] = None,
-    ):
-        dataset = HuggingFaceDataset(
-            dataset_name_or_path, **load_dataset_kwargs if load_dataset_kwargs else {}
-        )
-        data_loader: Union[HuggingFaceDataLoader, HuggingFaceLocalDataLoader] = get_hf_dataloader(
-            dataset
-        )
-        transformation: Union[
-            Transformation[datasets.DatasetDict, Corpus], Transformation[Corpus, Corpus]
-        ]
-        transformation = ColumnCorpusTransformation(
-            input_column_name=input_column_name,
-            target_column_name=target_column_name,
-            datasets_path=datasets_path,
-        )
-        if sample_missing_splits:
+        if self.downsample_splits:
             transformation = transformation.then(
-                SampleSplitsFlairCorpusTransformation(*sample_missing_splits, seed=seed)
+                DownsampleFlairCorpusTransformation(
+                    *self.downsample_splits,
+                    stratify=self.downsample_splits_stratification,
+                    seed=self.seed
+                )
             )
 
-        if ignore_test_subset:
-            transformation = transformation.then(DropSubsetFlairCorpusTransformation(subset="test"))
-        transformation = transformation.persisting(FlairConllPersister(path=persist_path))
-        super().__init__(dataset, data_loader, transformation)
+        transformation = transformation.persisting(self.persister)
+
+        return dataset, data_loader, transformation
+
+
+class FlairTextClassificationPreprocessingPipeline(FlairPreprocessingPipeline):
+    def _setup(self) -> None:
+        self.persister = FlairPicklePersister(self.persist_path)
```

Comment:
If we define `FlairPreprocessingPipeline` as a dataclass we can do the same here and set the persister as a class attribute. Then we are refactor the _setup and _setup_transformation methods as stated in the previous comment.